datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java

```diff
@@ -1007,6 +1007,10 @@ private void validateShape(SdxClusterShape shape, String runtime, DetailedEnviro
                         + MEDIUM_DUTY_REQUIRED_VERSION + " and not " + runtime);
             }
         } else if (SdxClusterShape.ENTERPRISE.equals(shape)) {
+            if (entitlementService.enterpriseSdxDisabled(Crn.safeFromString(environment.getCreator()).getAccountId())) {
```

Comment:
pls use `ThreadBasedUserCrnProvider.getAccount()`. using getCreator is an antipattern and the method is also deprecated

Reply:
fyi: in case of using `ThreadBasedUserCrnProvider`, for unit test you also need to use it like this: `ThreadBasedUserCrnProvider.doAs(SAMPLE_USER_CRN, () -> underTest.method())`, example: https://github.com/hortonworks/cloudbreak/blob/master/environment/src/test/java/com/sequenceiq/environment/environment/service/EnvironmentStopServiceTest.java#L72